lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -480,6 +485,173 @@ def post(self, request):
             branch = request.POST.get('repo_branch', '').strip().replace(' ', '').replace(';', '')
             self.msg += self.get_course_from_git(gitloc, branch)
 
```

Comment:
Please split this into several functions, it is very hard to read such long sheets of code.